libs/auth/arguments_test.go

```diff
@@ -0,0 +1,77 @@
+package auth
+
+import (
+	"testing"
+
+	"github.com/databricks/databricks-sdk-go/credentials/u2m"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestToOAuthArgument(t *testing.T) {
+	tests := []struct {
+		name      string
+		args      AuthArguments
+		wantHost  string
+		wantError bool
+	}{
+		{
+			name: "workspace with no scheme",
+			args: AuthArguments{
+				Host: "my-workspace.cloud.databricks.com",
+			},
+			wantHost: "https://my-workspace.cloud.databricks.com",
+		},
+		{
+			name: "workspace with https",
+			args: AuthArguments{
+				Host: "https://my-workspace.cloud.databricks.com",
+			},
+			wantHost: "https://my-workspace.cloud.databricks.com",
+		},
+		{
+			name: "account with no scheme",
+			args: AuthArguments{
+				Host:      "accounts.cloud.databricks.com",
+				AccountID: "123456789",
+			},
+			wantHost: "https://accounts.cloud.databricks.com",
+		},
+		{
+			name: "account with https",
+			args: AuthArguments{
+				Host:      "https://accounts.cloud.databricks.com",
+				AccountID: "123456789",
+			},
+			wantHost: "https://accounts.cloud.databricks.com",
+		},
+		{
+			name: "workspace with query parameter",
+			args: AuthArguments{
+				Host: "https://my-workspace.cloud.databricks.com?o=123456789",
+			},
+			wantHost: "https://my-workspace.cloud.databricks.com",
+		},
```

Comment:
Can you add a test case that includes an HTTP path?

That is also likely to be copied from the browser but doesn't make sense here.